File: docs/current_docs/agents/snippets/coding-agent/php/src/CodingAgent.php
<?php

declare(strict_types=1);

namespace DaggerModule;

use Dagger\Attribute\DaggerFunction;
use Dagger\Attribute\DaggerObject;
use Dagger\Attribute\Doc;
use Dagger\Container;

use function Dagger\dag;

#[DaggerObject]
#[Doc('A generated module for CodingAgent functions')]
class CodingAgent
{
    #[DaggerFunction]
    #[Doc('Write a Go program')]
    public function goProgram(string $assignment): Container
    {
        $workspace = dag()->toyWorkspace();
        $environment = dag()->env()
            ->withToyWorkspaceInput("before", $workspace, "these are the tools to complete the task")
            ->withStringInput("assignment", $assignment, "this is the assignment, complete it")
            ->withToyWorkspaceOutput("after", "the ToyWorkspace with the completed assignment");
        return dag()
            ->llm()
            ->withEnv($environment)
            ->withPrompt("
            You are an expert go programmer. You have access to a workspace.
			Use the default directory in the workspace.
			Do not stop until the code builds.")
            ->env()
            ->output("after")
            ->asToyWorkspace()
            ->container();
    }
}

File: docs/current_docs/agents/snippets/coding-agent/java/src/main/java/io/dagger/modules/codingagent/CodingAgent.java
package io.dagger.modules.codingagent;

import static io.dagger.client.Dagger.dag;

import io.dagger.client.Container;
import io.dagger.client.ToyWorkspace;
import io.dagger.client.Env;
import io.dagger.module.annotation.Function;
import io.dagger.module.annotation.Object;

/** CodingAgent main object */
@Object
public class CodingAgent {
  /** Write a Go program */
  @Function
  public Container goProgram(String assignment) {
    ToyWorkspace workspace = dag.toyWorkspace();
    Env environment = dag.env()
        .withToyWorkspaceInput("before", workspace, "these are the tools to complete the task")
        .withStringInput("assignment", assignment, "this is the assignment, complete it")
        .withStringOutput("after", "the ToyWorkspace with the completed assignment");
    return dag()
      .llm()
      .withEnv(environment)
      .withPrompt("""
        You are an expert go programmer. You have access to a workspace.
        Use the default directory in the workspace.
        Do not stop until the code builds.
        """)
      .env()
      .output("after")
      .asToyWorkspace()
      .container();
  }
}
